dashbuilder-shared/dashbuilder-dataset-shared/src/main/java/org/dashbuilder/dataset/engine/function/MaxFunction.java

```diff
@@ -38,7 +38,10 @@ public Object aggregate(List values) {
             return null;
         }
 
-        List sorted = (List) values.stream().sorted().collect(Collectors.toList());
+        List sorted = (List) values.stream()
+                .filter(i -> i != null)
+                .sorted().collect(Collectors.toList());
```

Comment:
Edit: i might be wrong about what's causing the performance difference (it might be collecting the sorted data `toList()`), but the difference is noticeable with larger data sets (100k+ items) especially when multiple aggregation functions for different columns are used.